optimism/src/keccak/constraints.rs

```diff
@@ -26,8 +30,38 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
     }
 
     fn constraints(&mut self) {
-        todo!();
         // CORRECTNESS OF FLAGS
+        {
+            // TODO: remove redundancy if any
+
+            // Booleanity of sponge flags
+            {
+                // Absorb is either true or false
+                self.constrain(Self::boolean(self.absorb()));
```

Comment:
Follow-up: I think the operations defined in BoolOps overlap with some from the interpreter env.

Reply:
Opened https://github.com/o1-labs/proof-systems/issues/1609

Reply:
Probably, they were created in parallel